src/qdmi/Device.cpp

```
@@ -1,174 +1,168 @@
  * The device maintains:
  * - Global state (idle/busy based on running jobs)
  * - Session registry for lifecycle management
  * - Random number generation for unique job IDs
  */
-Device::Device() {
-  name_ = "AWS Braket Device";
-  provider_ = "AWS";  // AWS is the cloud provider
-  deviceArn_ = "";    // ARN (Amazon Resource Name) identifies the specific quantum device
-  deviceType_ = "QPU"; // Quantum Processing Unit
-  qubitsNum_ = 0;      // Will be populated when device capabilities are queried
+Device::Device()
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: name_, provider_, deviceArn_, deviceType_, sessionsMutex_, rngMutex_

(cppcoreguidelines-pro-type-member-init)